peps/pep-0807.rst

```diff
@@ -0,0 +1,398 @@
+
+   For the above example, the path component is
+   ``/legacy/``.
+
+3. The uploading client takes the SHA2-256 hash of the path component,
+   producing the *discovery key*.
+
+   For the above example, the discovery key is
+   ``af030c06750716b1b35852298fe852b90def13dcbd012a5fe5148470f1206bfc``. [#fn-hash]_
+
+4. The uploading client constructs a *discovery URL* by taking the
+   scheme and authority components (as defined in :rfc:`3986`)
+   of the upload URL and appending ``/.well-known/pytp/``
+   and the discovery key.
+
+   For the above example, the discovery URL is
+   ``https://upload.example.com/.well-known/pytp/af030c06750716b1b35852298fe852b90def13dcbd012a5fe5148470f1206bfc``.
+
+5. The uploading client performs an HTTP GET request to the discovery URL.
+
+6. The server responds with a ``200 OK`` status code and a body
+   containing a JSON object if the index supports Trusted Publishing
+   for the given upload URL. The JSON object **MUST** contain the following
+   fields:
+
+   - ``audience-endpoint``: a string containing the URL of the OIDC
+     audience endpoint to be used during token exchange.
+   - ``token-mint-endpoint``: a string containing the URL of the
+     token minting endpoint to be used during token exchange.
+
+   For the above example, a valid response body would be:
+
+   .. code-block:: json
+
+       {
+            "audience-endpoint": "https://upload.example.com/_/oidc/audience",
+            "token-mint-endpoint": "https://upload.example.com/_/oidc/mint-token"
+       }
+
+If the server does not support Trusted Publishing for the given
+upload URL, it **MUST** respond with a ``404 Not Found`` status code.
+When responding with a ``404 Not Found``, the server **SHOULD NOT**
+include a response body. If a response body is included, it **MUST**
+be ignored by the client.
+
+Servers **MAY** additionally respond with any other standard HTTP
+error code in the 400 or 500 range to indicate an error condition.
+
+Non-``200 OK``, non-``404 Not Found`` responses **MAY** include a body which,
+if present, **MUST** be a JSON object containing an
+:ref:`Error Response <error-responses>`.
+
+.. _token-exchange:
+
+Trusted Publishing Token Exchange
+---------------------------------
+
+Once an uploading client has performed a successful :ref:`discovery <discovery>`
+flow, it can proceed to perform the actual Trusted Publishing token exchange.
+
+Token exchange occurs in three steps:
+
+1. The uploading client uses the *audience endpoint* obtained
+   during discovery to ask the index for its expected OIDC audience.
+2. The uploading client uses the expected audience to obtain an
+   appropriately bound *identity credential* from the Trusted Publishing
+   provider being used (i.e. the CI/CD or cloud provider that the upload
+   is being performed from). The details of this step are provider-specific,
+   and are out of scope for this PEP. [#fn-oidc]_
+3. The uploading client uses the *mint token endpoint* obtained
```

Comment:
The JSON uses "token mint endpoint". Changing the word order probably doesn't matter, but threw me for a second.

Reply:
Yeah, this was from https://github.com/python/peps/pull/4597#discussion_r2363798207 -- the existing endpoint on PyPI is `mint-token`, and I don't have strong opinion about the order of these words. Maybe "token minting" to make it more prosaic...